### DISCUSSION TITLE
504 error

### DISCUSSION THREAD

**Discussion by NursultanMRX** [vercel/community#6853] [Help] [answered]
### Summary

Why am I getting a 504 error in my nextjs project? I'm using in OpenAI API

### Example

Request failed with status code 504

### Steps to Reproduce

![Screenshot from 2024-05-15 20-06-29](https://github.com/vercel/community/assets/121639994/e29f2c44-1d72-44ac-9e97-5e799435c4fe)

**Comment by swarnava** [comment, marked as answer]
Hi,

The FUNCTION_INVOCATION_TIMEOUT (504) error occurs when a function invocation takes longer than the [allowed execution time](https://vercel.com/docs/limits/overview#serverless-function-execution-timeout). This could be due to an error within the function itself, a slow network call, or an issue with the environment in which the function is running. You can learn more here https://vercel.com/docs/errors/FUNCTION_INVOCATION_TIMEOUT#troubleshoot

**Reply by amyegan** [reply]
Apps using LLMs often benefit from streaming. More info on that at https://vercel.com/docs/functions/streaming
